api/src/pcapi/core/educational/models.py

```diff
@@ -129,9 +144,12 @@ def set_image(
         ratio: ImageRatio = ImageRatio.PORTRAIT,
         keep_original: bool = False,
     ) -> None:
+        old_id = self.imageId
         if self.hasImage:  # pylint: disable=using-constant-test
             self.delete_image()
-        self.imageCrop = crop_params.__dict__
+
+        self.imageId = self._generate_new_image_id(old_id)
```

Comment:
On pourrait utiliser `uuid4()` ? Ça fait le même travail que `random()`, mais sans risque de collision, non ?

```suggestion
        self.imageId = uuid.uuid4()
```

Reply:
je ne sais aps comment va se comporter UUID4 avec k8s (a cause du materiel et du kernel partagé).  

La façon de faire actuelle garantis l'absence de collision. L'avantage d'avoir une méthode distincte pour la génération de l'imageId est de permettre aux différentes classes héritant de ce mixin de le reimplémenter selon leurs besoins.

Reply:
(l'uuid4 ne garantis pas l'absence de collisions mais la rend tres improbable comme tout grand nombre aléatoire)